base/boot.jl

```diff
@@ -747,6 +747,7 @@ Int64(x::Ptr) = Int64(UInt32(x))
 UInt64(x::Ptr) = UInt64(UInt32(x))
 end
 Ptr{T}(x::Union{Int,UInt,Ptr}) where {T} = bitcast(Ptr{T}, x)
+Ptr{T}() where {T} = Ptr{T}(0)
```

Comment:
What if `T(0)` doesn't make sense?

Reply:
Hmm, doesn't it always make sense to just set a pointer's bits to zero as a safe initialization? Is there a case I'm not thinking of?

Reply:
@ararslan the pointer is 0, not that value.

Maybe use `C_NULL` instead of `0` to make it more obvious?

Reply:
Oh you're right, I misinterpreted how that constructor acts.

Reply:
Oh smart.